### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by dshanske**
This PR removes 2 duplicate functions now in a different file and it splits out form to json conversion into its own function.

### CODE REVIEW

File: includes/class-micropub-endpoint.php [hunk 1]
```diff
@@ -895,21 +891,7 @@ protected static function load_input() {
 		} elseif ( ! $content_type ||
 			   'application/x-www-form-urlencoded' === $content_type ||
 			   'multipart/form-data' === $content_type ) {
-			static::$input = array();
```
Comment (snarfed) [inline includes/class-micropub-endpoint.php]: this got lost in the move. it should be added back at the top of `form_to_json`, right?

**Comment by snarfed** [comment]
hi @dshanske! i don't see how the `indieauth_*_header` methods are duplicated. i only see them defined once, which you're remove in this PR.

you don't mean they're duplicated in the indieauth plugin, do you?

**Comment by dshanske** [comment]
They are in the authorize class also

File: includes/class-micropub-endpoint.php (hunk 1 above)
Comment (dshanske) [reply includes/class-micropub-endpoint.php]: It should be $input=array() but yes

**Comment by snarfed** [comment]
> They are in the authorize class also

ah, yes, thanks. i was naively looking for exact name matches, but they don't have the `indieauth_` prefix there. got it.

**Comment by snarfed** [comment]
if you want to remove those methods, you need to remove their `add_action` and `add_filter`s too:

https://github.com/dshanske/wordpress-micropub/blob/800fb11206191c75fa637a5188d1fe574882d45f/includes/class-micropub-endpoint.php#L40-L46

**Comment by dshanske** [comment]
@snarfed Fixed now.

**Review by snarfed — APPROVED** [review]
thanks!

**Pull request merged**